crates/consensus/src/transaction/mod.rs

```diff
@@ -47,20 +47,16 @@ pub trait Transaction: any::Any + Send + Sync + 'static {
     /// Get `gas_limit`.
     fn gas_limit(&self) -> u64;
 
-    /// Get `gas_price`.
-    fn gas_price(&self) -> Option<u128>;
```

Comment:
what's our motivation for removing this? I can see how it's redundant from execution perspective, though it is an actual separate property of transaction which is important from RPC PoV

I think it's pretty much a question on how we expect this trait to be consumed and whether separate getters might be useful

Reply:
the gas price field is never used on tx types since max fee per gas was introduced